pdns/bpf-filter.cc

```diff
@@ -320,6 +322,25 @@ BPFFilter::BPFFilter(std::unordered_map<std::string, MapConfiguration>& configs,
     throw std::runtime_error("Unsupported eBPF map format, the current internal implemenation only supports the legacy format");
   }
 
+  struct rlimit old_limit;
+  const size_t new_limit_size = 1073741824;
+
+  if (getrlimit(RLIMIT_MEMLOCK, &old_limit) != 0) {
+    throw std::runtime_error("Unable to get memory lock limit: " + stringerror());
+  }
+
+  /* Check if the current soft memlock limit is 64k */ 
+  if (old_limit.rlim_cur < 67108865) {
+    struct rlimit new_limit;
+    new_limit.rlim_cur = new_limit_size; /* Increase soft limit to 1024k */
+    new_limit.rlim_max = new_limit_size; /* Increase hard limit to 1024k */
+  
+    if (setrlimit(RLIMIT_MEMLOCK, &new_limit) != 0) {
+      errlog("Unable to raise the maximum amount of locked memory for eBPF from %d to %d, consider raising RLIMIT_MEMLOCK or setting LimitMEMLOCK=infinity in the systemd unit: %s", old_limit.rlim_cur, new_limit.rlim_cur, stringerror());
+    }
+    infolog("The current limit of locked memory (soft: %d, hard: %d) is too low for eBPF, trying to raise it to %d", old_limit.rlim_cur, old_limit.rlim_max, new_limit_size);
```

Comment:
two times wrong format specifier `%d` for `rlim_t`.

Reply:
I'm afraid that one is on me, sorry about that. It's not clear from POSIX what the correct specifier would be, though, but I guess at least `%u` would be better than `%d`:

> rlim_t
    Unsigned integer type used for limit values.

Reply:
`%llu` and a cast to `unsigned long long` of the value would be safe.

Reply:
Scratch that, `infolog` does not use `printf` format specifiers but a completely different mechanism. `%d` is fine as the output is written as `stream << value` in the end.